### ISSUE TITLE
MQTT: not supported for MQTT topics

### ISSUE THREAD

**Issue by tonobo** [nats-io/nats-server#4836]
### Observed behavior

```
[7] 2023/11/30 20:04:43.158454 [ERR] 10.0.0.15:49368 - mid:1834 - "z2m" - characters ' ' and '.' not supported for MQTT topics
[7] 2023/11/30 20:04:44.170766 [ERR] 10.0.0.15:49384 - mid:1835 - "z2m" - characters ' ' and '.' not supported for MQTT topics
[7] 2023/11/30 20:04:45.191075 [ERR] 10.0.0.15:49396 - mid:1836 - "z2m" - characters ' ' and '.' not supported for MQTT topics
[7] 2023/11/30 20:04:46.204181 [ERR] 10.0.0.15:49400 - mid:1837 - "z2m" - characters ' ' and '.' not supported for MQTT topics
[7] 2023/11/30 20:04:47.209087 [ERR] 10.0.0.15:49404 - mid:1838 - "z2m" - characters ' ' and '.' not supported for MQTT topics
```

### Expected behavior

expected to not reject the message

### Server and client version

Server: nats-helm 1.1.5
Client: zigbee2mqtt

### Steps to reproduce

zigbee2mqtt seems to emit unprocessable mqtt mesages, i tried to switch from mosquitto over to nats mqtt without success. Homeassistant stuff works ootb, but z2m is causing trouble. I thought orignally it is due to mqtt.js, but relaying the mqtt traffic through working mosquitto is causing the same issues. I'm not very familar with mqtt protocol stuff, not sure if nats is doing something wrong with mqtt or mqtt is being abused by z2m. Maybe someone can help me :D

**Comment by tonobo** [comment]
by dropping all the spaces from device names, nats seem to work

**Comment by kozlovic** [comment]
Yes, NATS Server cannot support spaces in MQTT topics because spaces are not allowed in NATS subjects. By the way, the error message should be fixed because since v2.10.0, we do support the '.' character.
